src/test/java/org/commcare/backend/suite/model/test/CaseClaimModelTests.java

```diff
@@ -35,6 +45,43 @@ public void testRemoteQueryDatum() throws Exception {
         Assert.assertTrue("Didn't find Remote Query datum definition", datum instanceof RemoteQueryDatum);
     }
 
+    @Test
+    public void testPopulateItemsetChoices__inputReference() throws Exception {
+        testPopulateItemsetChoices(ImmutableMap.of("state", "ka"), ImmutableList.of("bang"));
+    }
+
+    @Test
+    public void testPopulateItemsetChoices__emptyInput() throws Exception {
+        testPopulateItemsetChoices(Collections.emptyMap(), Collections.emptyList());
+    }
+
+    private void testPopulateItemsetChoices(Map<String, String> userInput, List<String> expected)
+            throws Exception {
+        MockApp mApp = new MockApp("/case_claim_example/");
+
+        SessionWrapper session = mApp.getSession();
+        session.setCommand("patient-search");
+
+        VirtualDataInstance districtInstance = buildDistrictInstance();
+        EvaluationContext context = session.getEvaluationContext().spawnWithCleanLifecycle(ImmutableMap.of(
```

Comment:
Quite cool that we can now inject data instances without relying on loading them through restore.